sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/pipeline/TelemetryItemExporter.java

```diff
@@ -122,8 +122,6 @@ private CompletableResultCode maybeAddToActiveExportResults(List<CompletableResu
         for (CompletableResultCode result : results) {
             result.whenComplete(() -> activeExportResults.remove(result));
         }
-
-        return CompletableResultCode.ofSuccess();
```

Comment:
this was the problem, it should have been returning `CompletableResultCode.ofAll(results)` (which is the same as above, and so also simplified by moving it to the caller)

Reply:
how did you catch this? i wonder if we need to update tests to catch it.

Reply:
I caught it during manual testing

we have a smoke test in the Java agent for testing JVM shutdown hook: https://github.com/microsoft/ApplicationInsights-Java/tree/main/smoke-tests/apps/SystemExit

we'd need to think how to test it in this repo